deploy.sh

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+set -e # Exit with nonzero exit code if anything fails
+
+# From https://gist.github.com/domenic/ec8b0fc8ab45f39403dd
+
+echo "[TRACE] TRAVIS_BRANCH: $TRAVIS_BRANCH"
+echo "[TRACE] TRAVIS_PULL_REQUEST_BRANCH: $TRAVIS_PULL_REQUEST_BRANCH"
+echo "[TRACE] TRAVIS_PULL_REQUEST: $TRAVIS_PULL_REQUEST"
+
+SOURCE_BRANCH="$TRAVIS_PULL_REQUEST_BRANCH"
+TARGET_BRANCH="$TRAVIS_PULL_REQUEST-$TRAVIS_PULL_REQUEST_BRANCH"
+
+if [ "$TRAVIS_PULL_REQUEST" = "false" -a "$TRAVIS_BRANCH" != "master" ]; then
+   echo "[ABORT] We're in a push build (not in master nor in a pull request), so exiting. "
```

Comment:
I'd prefer that this at least validates the push build to check it looks okay rather than exiting without doing anything. Possibly I've not understood the state here though.

Reply:
push build are only triggered when one pushes against a branch. If that branch is part of the pull request, a second build (aka a PR build) will be triggered as well. So, for each new pull request or new commit in a pull request, the script is invoked twice but will only trigger the ant build once. In practice, I don't think  push build matters that except if it's the master branch of course. PR build is the right place to catch errors imho.

Reply:
OK, I suppose the only thing that matters is that the build will succeed after the merge.